Source/SpectrumAnalysis/KTSequentialTrackFinder.cc

```diff
@@ -329,225 +341,312 @@ namespace Katydid
-                     newLine.AddPoint(*pointIt);
-                     (this->*fCalcSlope)(newLine);
-                     fActiveLines.push_back(newLine);
-                     match = true;
-                 }
-             }
-         }
-         return true;
-     }
+        uint64_t acqID = kdTreeData.GetAcquisitionID();
+
+        // We need to be able to detect when we've moved from slice to slice
+        // So we define a threshold for delta-t, because time values might not be the exact same due to floating-point uncertainty.
+        // We'll use the bin width, which is tiny relative to the slice size and stride.
+        // 3-times the bin width should be large enough compared to uncertainty on the slice time and smaller than any reasonable stride.
+        double deltaTThreshold = 3. * fTimeBinWidth;
+
+        for (unsigned iComponent = 0; iComponent < nComponents; ++iComponent)
+        {
+            unsigned nBins = fMaxBin - fMinBin + 1;
+
+            const KTKDTreeData::SetOfPoints& allPoints = kdTreeData.GetSetOfPoints(iComponent);
+
+            // this set will collect the discriminated points sorted by power
+            STFDiscriminatedPowerSortedPoints points;
+
+            double lastTime = -1000.;
+            for (auto pointIt = allPoints.begin(); pointIt != allPoints.end(); ++pointIt)
+            {
+                double time = pointIt->fCoords[0] * kdTreeData.GetXScaling();
+                double freq = pointIt->fCoords[1] * kdTreeData.GetYScaling();
+
+                // check if we've moved onto the next slice and if we have points collected already
+                if (time - lastTime > deltaTThreshold && ! points.empty())
+                {
+                    // this point is on the next slice; process these points then reset the points set
+                    KTDEBUG( stflog, "Collected " << points.size() << " points");
+
+                    // Loop over the high power points
+                    this->LoopOverHighPowerPoints(points, acqID, iComponent);
+
+                    // we're done with those points
+                    points.clear();
+                }
+
+                // check if it's not noise, and that we're within the frequency bounds
+                if (! pointIt->fNoiseFlag && freq > fMinFrequency && freq < fMaxFrequency)
+                {
+                    points.emplace(pointIt, time, freq, kdTreeData.GetXScaling());
+                }
+
+                lastTime = time;
+            }
+
+            if (! points.empty())
+            {
+                KTDEBUG( stflog, "Collected "<<points.size()<<" points");
+
+                // Loop over the high power points
+                this->LoopOverHighPowerPoints(points, acqID, iComponent);
+            }
+        }
+        return true;
+    }
+
+    bool KTSequentialTrackFinder::LoopOverHighPowerPoints(KTPowerSpectrum& slice, STFDiscriminatedPowerSortedPoints& points, uint64_t acqID, unsigned component)
```

Comment:
We are not using this method any more and with a non-normalized power spectrum slice it produces very biased (frequency dependent) results. Future me will probably remove it one day.

Reply:
Sounds good to me.  I'm all for cleaning up unused code (though I often fail at that ideal)